packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -269,32 +269,11 @@ const App = (props) => {
         history.push(path)
     }
 
-    // A major pain point for the current development of Storefront Preview is that we don't have
-    // static URLs on deployed sites. The path /mobify/bundle/123/static/storefront-preview.js
-    // changes to a different number with every deployment. Once we figure out the static URL we can
-    // update it for production, but will still need to come up with something for staging...
-
-    const storefrontPreviewClientScript =
-        process.env.NODE_ENV === 'development'
-            ? 'http://localhost:4000/mobify/bundle/development/static/storefront-preview.js'
-            : // Gotta update this manually for now... :(
-              // : 'https://runtime.commercecloud.com/storefront-preview.js'
-              null
-
-    if (!storefrontPreviewClientScript) {
-        // Deliberately aggressive because this branch exists solely for Storefront Preview development
-        return (
-            <Box className="sf-app">
-                Hey! You forgot to set a Storefront Preview client script URL!
-            </Box>
-        )
-    }
+    const StorefrontPreview = useStorefrontPreview()
```

Comment:
It is kinda odd to call a hook that returns a component. Why don't we just create it as a StorefrontPreview with two optional props custonisation and enabled, then we can render it as
```
import {useStorefrontPreview} from '@salesforce/pwa-kit-react-sdk/ssr/universal/hooks'

return (
   <Box className="sf-app" {...styles.container}>
         <StorefrontPreview />

   </Box>

)
```

Reply:
In order to have `forceUpdate` work properly, it cannot be defined inside `<StorefrontPreview>` component. (If it is, it will only rerender the `<StorefrontPreview>` component, rather than the entire app.) We want to avoid defining `forceUpdate` in the root app, so we have to use a hook in order for `forceUpdate` to work as desired. But we also want to inject a script tag, and the React-y way to do that is to return a component!

But I've realized that we're injecting a script tag for something that exists outside react, so there's no impact for doing plain DOM manipulation. So now we have a pure hook, and only 2 lines of change neeed! (import + `useStorefrontPreview()`)